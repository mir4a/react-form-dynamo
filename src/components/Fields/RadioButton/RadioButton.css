.RadioButton {
  margin-bottom: 25px;
}