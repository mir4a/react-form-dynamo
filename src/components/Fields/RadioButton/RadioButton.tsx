import React, { memo, useState, useCallback } from 'react';
import Label from '../Label';
import './RadioButton.css';

export type RadioButtonType = {
  label: string;
  value: string;
};

interface Props {
  initialValue?: string;
  label: string;
  name?: string;
  onChange?: Function;
  items: RadioButtonType[];
}

/**
 * whether values: [{label: string, value: string}]
 */

export default memo(function RadioButton(props: Props) {
  const { initialValue, label, items, name, onChange } = props;
  const [selectedValue, setSelectedValue] = useState(initialValue);
  const handleOnChange = useCallback(
    (event) => {
      const value = event.target.value;

      setSelectedValue(value);
      onChange && onChange(value);
    },
    [onChange, setSelectedValue],
  );

  return (
    <div className="RadioButton">
      <Label label={label} className="Label__block" />
      {items.map((item) => {
        const checked = String(selectedValue) === String(item.value);

        return (
          <Label label={item.label} key={item.value} className="Label__block">
            <input
              type="radio"
              name={name}
              value={item.value}
              checked={checked}
              onChange={handleOnChange}
            />
          </Label>
        );
      })}
    </div>
  );
});
