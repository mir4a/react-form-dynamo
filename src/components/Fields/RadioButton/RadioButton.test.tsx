import React from 'react';
import renderer from 'react-test-renderer';
import RadioButton from './RadioButton';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <RadioButton
        items={[
          { label: 'a', value: '1' },
          { label: 'b', value: '2' },
        ]}
        initialValue="1"
        label="Radio Buttons"
        onChange={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
