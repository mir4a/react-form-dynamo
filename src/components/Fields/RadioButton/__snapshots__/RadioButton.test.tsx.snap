// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="RadioButton"
>
  <label
    className="Label Label__block"
  >
    <span
      className="Label-text"
    >
      Radio Buttons
    </span>
  </label>
  <label
    className="Label Label__block"
  >
    <input
      checked={true}
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <span
      className="Label-text"
    >
      a
    </span>
  </label>
  <label
    className="Label Label__block"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <span
      className="Label-text"
    >
      b
    </span>
  </label>
</div>
`;
